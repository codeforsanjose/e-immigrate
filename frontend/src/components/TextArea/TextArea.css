.TextArea {
    border: 2px solid var(--color-foggyBlue);
    border-radius: 10px;
    font-size: 1rem;
    padding: 10px 12px;
    width: 100%;
    outline: none;
}

.TextArea[touched='true']:focus {
    border: 2px solid #0072ff;
}



.RequiredError {
    display: none;
    color: var(--color-secondary);
}
