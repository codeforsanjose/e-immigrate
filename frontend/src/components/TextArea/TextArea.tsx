import React from 'react';
import './TextArea.css';
import { CommonComponentProps } from '../../types/CommonComponentProps';

type TextAreaProps = CommonComponentProps;
export function TextArea(props: TextAreaProps) {
    const {
        q, 
        bindField, 
        collectAnswer, 
        content,
    } = props;

    return (
        <>
            <textarea
                rows={4}
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)}
                onChange={(e) => collectAnswer(q.slug, e.target.value)} />
            <div className="RequiredError">*{content.errorMessage}</div>
        </>
    );
}
