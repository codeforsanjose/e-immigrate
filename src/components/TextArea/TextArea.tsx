import React from 'react';
import './TextArea.css';
import { QData, BindFieldFunction, CollectAnswerFunction, ReactSetter } from '../../types/common';

type TextAreaProps = {
    attributes: {
        q: QData;
        bindField: BindFieldFunction;
        collectAnswer: CollectAnswerFunction;
        content: {
            errorMessage: string;
        };
    }
};
export function TextArea(props: TextAreaProps) {
    
    const {
        attributes: { 
            q, 
            bindField, 
            collectAnswer, 
            content,
        },
     } = props;

    return (
        <>
            <textarea
                rows={4}
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)}
                onChange={(e) => collectAnswer(q.slug, e.target.value)} />
            <div className="RequiredError">*{content.errorMessage}</div>
        </>
    );
}
