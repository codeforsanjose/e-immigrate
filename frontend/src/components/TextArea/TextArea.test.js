import React from 'react';
import TextArea from './TextArea';

import renderer from 'react-test-renderer';

it('TextArea renders correctly', () => {
    const tree = renderer
        .create(
            <TextArea
                q={{}}
                answers={[]}
                values={[]}
                bindField={() => {}}
                showFollowUp={{}}
                setShowFollowUp={{}}
                collectAnswer={{}}
                content={{}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
