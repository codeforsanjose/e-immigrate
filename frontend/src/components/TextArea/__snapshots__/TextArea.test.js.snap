// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders correctly 1`] = `
Array [
  <textarea
    className="TextInput"
    onChange={[Function]}
    rows="4"
  />,
  <div
    className="RequiredError"
  >
    *
  </div>,
]
`;
